runtime/fundamentals/modules.md

```diff
@@ -0,0 +1,450 @@
+
+It is possible to specify a version range for the package you are importing.
+This is done using the `@` symbol followed by a version range specifier, and
+follows the [semver](https://semver.org/) versioning scheme.
+
+```bash
+deno add @scopename/mypackage           # latest version
+deno add @scopename/mypackage@16.1.0    # exact version
+deno add @scopename/mypackage@^16.1.0   # latest patch version 16.x
+deno add @scopename/mypackage@~16.1.0   # latest version that doesn't increment the first non-zero portion
+```
+
+## Importing by URL
+
+Deno also supports import statements that reference URLs, either directly:
+
+```js
+import { Application } from "https://deno.land/x/oak/mod.ts";
+```
+
+or part of your `deno.json` import map:
+
+```json
+{
+  "imports": {
+    "oak": "https://deno.land/x/oak/mod.ts"
+  }
+}
+```
+
+Supporting URL imports enables us to support the following JavaScript CDNs, as
+they provide URL access to JavaScript modules:
+
+- [deno.land/x](https://deno.land/x)
+- [esm.sh](https://esm.sh)
+- [unpkg.com](https://unpkg.com)
+
+URL imports are useful if you have a small, often single file, Deno project that
+doesn't require any other configuration because you can avoid having a
+`Deno.json` file at all. It's not advised to use this style of import in larger
+applications as you may end up with version conflicts (where different files use
+different version specifiers).
+
+From a security perspective, the contents of files at URLs can change, so we do
+not generally recommend this approach for third-party components.
+
+URL imports remain supported, **but we recommend using a package registry for
+the best experience.**
+
+### Overriding URL imports
+
+The other situation where import maps can be very useful is to override URL
+imports in specific modules.
+
+Let's say you want to override a `https://deno.land/x/my-library@1.0.0/mod.ts`
+specifier that is used inside files coming from `https://deno.land/x/example/`
+to a local patched version. You can do this by using a scope in the import map
+that looks something like this:
+
+```json
+{
+  "imports": {
+    "example/": "https://deno.land/x/example/"
+  },
+  "scopes": {
+    "https://deno.land/x/example/": {
+      "https://deno.land/x/my-library@1.0.0/mod.ts": "./patched/mod.ts"
+    }
+  }
+}
+```
+
+:::note
+
+URL imports have no notion of packages. Only the import map at the root of your
+project is used. Import maps used inside URL dependencies are ignored.
+
+:::
+
+## Proxies
+
+Deno is able to handle network requests through a proxy server, useful for
+various reasons such as security, caching, or accessing resources behind a
+firewall. The runtime supports supports proxies for module downloads and the Web
+standard `fetch` API.
+
+Deno reads proxy configuration from environment variables: `HTTP_PROXY`,
+`HTTPS_PROXY` and `NO_PROXY`.
+
+In case of Windows, if environment variables are not found, Deno falls back to
+reading proxies from the registry.
```

Comment:
This seems out of place. The proxy configuration works for modules but also for `fetch`. 

Maybe it makes sense here if its also written in the env var reference docs.